src/autofill.js

```diff
@@ -5,8 +5,15 @@ import {createDevice} from './DeviceInterface'
 (() => {
     if (!window.isSecureContext) return false
     try {
-        const deviceInterface = createDevice()
-        deviceInterface.init()
+        const startupAutofill = () => {
+            if (document.visibilityState === 'visible') {
+                const deviceInterface = createDevice()
+                deviceInterface.init()
+            } else {
+                document.addEventListener('visibilitychange', startupAutofill, {once: true})
+            }
+        }
+        startupAutofill()
```

Comment:
This is what we discussed about not kickstarting autofill until the document is visible. It sets the listeners with `once: true` until the document is visible so no matter if the document starts to `prerender` or whatever. Until it's visible, it keeps calling itself.